io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly25/IoShelly25Impl.java

```diff
@@ -100,68 +116,86 @@ public void handleEvent(Event event) {
-			relay2IsOn = JsonUtils.getAsBoolean(relay2, "ison");
+	private void processHttpResult(JsonElement result, Throwable error) {
+		this._setSlaveCommunicationFailed(result == null);
+		Boolean relay1 = null;
+		Boolean overtemp1 = null;
+		Boolean overpower1 = null;
+		Boolean relay2 = null;
+		Boolean overtemp2 = null;
+		Boolean overpower2 = null;
 
+		try {
+			JsonElement jsonElement = JsonUtils.getAsJsonElement(result);
+			final var relays = JsonUtils.getAsJsonArray(jsonElement, "relays");
+			for (int i = 0; i < relays.size(); i++) {
+				final var relay = JsonUtils.getAsJsonObject(relays.get(i));
+				final var relayIson = JsonUtils.getAsBoolean(relay, "ison");
+				final var relayOverpower = JsonUtils.getAsBoolean(relay, "overpower");
+				final var relayOvertemp = JsonUtils.getAsBoolean(relay, "overtemperature");
+
+				if (i == 0) {
+					relay1 = relayIson;
+					overtemp1 = relayOvertemp;
+					overpower1 = relayOverpower;
+				} else if (i == 1) {
+					relay2 = relayIson;
+					overtemp2 = relayOvertemp;
+					overpower2 = relayOverpower;
+				}
+			}
 			this._setSlaveCommunicationFailed(false);
-
-		} catch (OpenemsNamedException | IndexOutOfBoundsException e) {
-			relay1IsOn = null;
-			relay2IsOn = null;
-			this.logError(this.log, "Unable to read from Shelly API: " + e.getMessage());
-			this._setSlaveCommunicationFailed(true);
+		} catch (OpenemsNamedException e) {
+			this.logDebug(this.log, e.getMessage());
 		}
-		this._setRelay1(relay1IsOn);
-		this._setRelay2(relay2IsOn);
+		// Sets the Fault Channels accordingly
+		this._setRelay1Overpower(overpower1);
+		this._setRelay2Overpower(overpower2);
+		this._setRelay1Overtemp(overtemp1);
+		this._setRelay2Overtemp(overtemp2);
+
+		this._setRelay1(relay1);
+		this._setRelay2(relay2);
+
 	}
 
 	/**
 	 * Execute on Cycle Event "Execute Write".
 	 */
 	private void eventExecuteWrite() {
-		try {
-			this.executeWrite(this.getRelay1Channel(), 0);
-			this.executeWrite(this.getRelay2Channel(), 1);
-
-			this._setSlaveCommunicationFailed(false);
-		} catch (OpenemsNamedException e) {
-			this._setSlaveCommunicationFailed(true);
+		for (int i = 0; i < this.digitalOutputChannels.length; i++) {
+			this.executeWrite(this.digitalOutputChannels[i], i);
 		}
 	}
 
-	private void executeWrite(BooleanWriteChannel channel, int index) throws OpenemsNamedException {
+	private void executeWrite(BooleanWriteChannel channel, int index) {
 		var readValue = channel.value().get();
 		var writeValue = channel.getNextWriteValueAndReset();
-		if (!writeValue.isPresent()) {
-			// no write value
+		if (writeValue.isEmpty()) {
 			return;
 		}
 		if (Objects.equals(readValue, writeValue.get())) {
-			// read value = write value
 			return;
 		}
-		this.shellyApi.setRelayTurn(index, writeValue.get());
+		final String url = this.baseUrl + "/relay/" + index + "?turn=" + (writeValue.get() ? "on" : "off");
+		this.httpBridge.get(url).whenComplete((t, e) -> {
```

Comment:
Should this not set `this._setSlaveCommunicationFailed(e != null);` like in the other Shelly implementations?